cpp/src/slice2cpp/Gen.cpp

```diff
@@ -908,20 +950,33 @@ Slice::Gen::generate(const UnitPtr& p)
     {
         normalizeMetaData(p, true);
 
+        // Forward declares proxies, classes and structs.
```

Comment:
I want to reduce the number of Visitors in a follow-up PR. I believe the following would work:

1. forward declarations for proxies, classes and structs; enums, sequences and dictionaries
2. proxies
3. structs + classes + exceptions + interfaces

Not sure about the StreamVisitor.